libs/rand/random.go

```diff
@@ -17,9 +16,8 @@ const (
 // They all utilize math/rand's prng internally.
 //
 // All of the methods here are suitable for concurrent use.
-// This is achieved by using a mutex lock on all of the provided methods.
+// Different randomness each instance
 type Rand struct {
-	sync.Mutex
 	rand *mrand.Rand
 }
 
```

Comment:
what do you think about removing grand completely?

Reply:
rand is thread-safe by default. In the case of `NewSource()`, we can still use grand. But if we don't need global randomness then it's fine.

Reply:
I don't see how that can be of use. Do you have something on your mind?